### PR TITLE
Docker build improvements

### PR DESCRIPTION
**Pull request by hamishfagg**
Currently we build each platform's docker image and push cache layers to a repo separately. This is because buildkit doesn't support building and pushing cache for multiple platforms at once.

The issue with this approach is that we need to push images suffixed with `-amd64` and `-arm64` to dockerhub before building a manifest that merges those two images. This is messy and confusing for users, and triples the number of tags we have in our dockerhub repo.

This PR uses another approach to overcome buildkit's shortcoming
- Build all images for all platforms and push the images but *not* cache
- Build each platform's images again (should now be cached) and don't push the images, but push cache instead

This approach means we can also continue with other jobs without waiting for cache to be pushed (which can take a while).

### CODE REVIEW

File: docker/docker-bake.hcl [hunk 1]
```diff
@@ -13,31 +13,35 @@ variable "IMAGE" {
...
 }
 variable "BRANCH" {
   default = "stable"
 }
-
-function "get_platform_tag" {
-  params = []
-  result = replace("${equal(PLATFORM, "") ? "" : "-"}${PLATFORM}", "linux/", "")
+variable "ECR_REPO" {
+  default = "454861456664.dkr.ecr.us-east-2.amazonaws.com"
```
Comment (lucas-koontz) [inline docker/docker-bake.hcl:26]: Is it ok to publish this info in this repository?

Comment (lucas-koontz) [reply docker/docker-bake.hcl:26]: NVM, it was already in the code.

**Review by lucas-koontz — APPROVED** [review]
LGTM!

File: docker/docker-bake.hcl (hunk 1 above)
Comment (hamishfagg) [reply docker/docker-bake.hcl:26]: Good catch but yeah, I checked :)

**Pull request merged**